ui/app/hooks/useDappAuthorizationCheck.js

```diff
@@ -0,0 +1,24 @@
+import { useSelector } from 'react-redux'
+import { getOriginOfCurrentTab, getPermittedAccountsByOrigin, getSelectedAddress, txDataSelector } from '../selectors'
+import { useMemo } from 'react'
+
+export function useDappAuthorizationCheck (address) {
+  const originOfCurrentTab = useSelector(getOriginOfCurrentTab)
+  const permittedAccountsByOrigin = useSelector(getPermittedAccountsByOrigin)
+  const selectedAddress = useSelector(getSelectedAddress)
+  const txData = useSelector(txDataSelector)
+
+  // Should this fallback to originOfCurrentTab? Not sure I understand this
+  const origin = txData?.origin || txData?.msgParams?.origin || originOfCurrentTab || undefined
```

Comment:
@Gudahtt i could use an extra set of eyes on this hook.

first, some confirmations have the origin at the top level of txData, some have it in msgParams, and I'm unsure if I should be using originOfCurrentTab here, it seems useful in the event this hook would be used outside of a confirmation screen (if I understand getOriginofCurrentTab's intent correctly)

Secondly, I'm unsure of the name, return values, etc for this hook. It's nice to have all of this logic bundled together for use in the various confirmation screens but I want to make sure it makes sense. Also if we do decide to throw the origin into the tooltip message, we might want to return the actual warning message from this hook? if not that, then return the origin itself?

Reply:
The "currentTab"-related stuff should probably be renamed to "activeTab" - that all refers to [this extension API feature](https://developer.chrome.com/extensions/activeTab), which lets you see the active tab when the browser action popup is triggered (e.g. when you click the MetaMask icon in the toolbar).

This is only set in the popup UI, so we only use it for popup-specific UI (like the connected status indicator). It shouldn't be used for confirmation screens at all, since those are typically in the notification popup, where there is no active tab.

I'll have to look a bit further into how this hook is being used, but I don't think there are any scenarios where we'd want to fallback from some other origin to the active tab origin. All transactions we receive from a dapp should have an origin associated with them - otherwise something is broken. I thought that even the send flow transactions had _something_ set as the origin :thinking:

Reply:
I'm a bit confused by the `dappIsAuthorizedForAccount` check - the dapp should _always_ be authorized for this account, otherwise it wouldn't be possible for us to get here. Is there a case you found where a dapp was able to suggest a transaction for an account they aren't permitted to interact with?

Reply:
Oh, I guess I just referenced one case: the send flow. Since any transactions created via the send flow are directly proposed by the user, not the dapp, it doesn't make sense to verify the origin is permitted for those. The interaction in that case is entirely independent of the dapp.

Reply:
hmm, I think I might have confused myself when trying to name this API after having written the logic. The check against the domain is really there to see if the request is being initiated by a Dapp the user has connected to in any of their accounts. Maybe I went to overboard with logic here and it's really as simple as 

```
const showWarning = selectedAccount !== fromAccount;
```

Reply:
bc, the only case where the fromAccount would be different from selected is in the event of a dapp initiated transaction

Reply:
I think it may be as simple as `const showWarning = selectedAccount !== fromAccount;` - even the send flow example I thought of earlier doesn't really matter - we can still show this warning if the "from" account differs from the selected account in the send flow, because that should rarely happen anyway.